wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && bash Miniconda3-latest-Linux-x86_64.sh -b && source ~/.bashrc && conda create -y -n biomol python=3.12.9 && conda activate biomol && conda install -y -c conda-forge biopython && conda install -y -c bioconda exonerate bedtools && python3 dna2protein.py contig_25.fa translated_nucleotide.fa && exonerate -m p2g --showtargetgff -q Arabidopsis_suecica_prot.fa.txt -t contig_25.fa -S F | egrep -w exon > pred_proteins_exonerate.gff && awk '$3 == "exon" {print $1"\t"($4-1)"\t"$5"\texon\t.\t"$7}' pred_proteins_exonerate.gff > prediction.bed && bedtools getfasta -fi contig_25.fa -bed prediction.bed -s -name > exons.fa && sed -e '2,$s/>.*//' exons.fa | grep -v '^$' > cds_exonerate.txt && python3 dna2protein.py cds_exonerate.txt pred_protein_exonerate.fa && cat pred_protein_exonerate.fa ab_initio_proteins.fa.txt > comparison_all.fa && t_coffee -in=comparison_all.fa -mode=expresso -blast=LOCAL -pdb_db=/db/pdb/derived_data_format/blast/2021-11-16/pdb_seqres.fa -evaluate_mode=t_coffee_slow -output=score_html clustalw_aln fasta_aln score_ascii phylip -maxnseq=150 -maxlen=2500 -case=upper -seqnos=off -outorder=input -run_name=result -multi_core=4 -quiet=stdout
